### PR TITLE
Frame technical plans as a way to make decisions

### PR DESCRIPTION
**Pull request by ashfurrow**
@dleve123 and I have been working for some time on defining a process for making technology decisions at Artsy. While the solution we landed on seems straightforward, I want to describe our process so that it's clear how we ended up here.

First, the rationale. Not all engineers at Artsy are familiar with how we make decisions (we saw this in recent Culture Amp surveys). There is a desire to know how we make decisions, and having a documented process is really important to make sure we build+maintain a culture where every engineer feels like they have a voice.

Daniel and I worked through a few different models, but none of them really suited our team. They were good frameworks, but introduced more and more process. This ran contrary to our Engineering Guiding Principle of [Minimum Viable Process](https://github.com/artsy/README/blob/master/culture/engineering-principles.md#minimal-viable-process). We realized that any process for making decisions will have significant overlap with our existing technical plans (see #245) so we decided to _lean into that overlap_. Rather than use two processes for technical planning and technical decision-making, we should use the existing technical planning process _for_ decision-making.

This is quite a shift, and will take time to socialize it. Daniel is going to run this through the paces on the Auctions team and, in the new year, I'm going to lead a Lunch & Learn discussion so we can retrospectively write a technical plan for decisions that we've already made. This is an ongoing process, and everyone's feedback is super-valuable.

To get us moving in the right direction, I've amended the (already-improved, thanks @joeyAghion!) technical choices doc to emphasize the usefulness of technical plans for making decisions. The documentation has a little extra pep so as to encourage engineers to write plans. I'll make a larger announcement on Slack and in Monday's standup.

I'm assigning this to @dleve123 because we worked on it together, and have requested a review from @xtina-starr because her input through the Peer Learning Working Group was super-valuable, ultimately pointing us in this direction.

### CODE REVIEW

**Comment by zephraph** [comment]
I think it would be good in some form to differentiate between when to use technical plan vs when to use an RFC.

**Comment by ashfurrow** [comment]
@zephraph makes sense! I think that might fit well within the FAQ section.

File: playbooks/technology-choices.md [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
+- If you have an idea,
+  [write a technical plan](https://www.notion.so/artsy/Technical-Plans-f94b206fcec54cee8b4d864e67d5b70f), then...
+- Share the plan with the team: bring it up
+  [at standup](https://github.com/artsy/README/blob/master/events/open-standup.md) in "cross-team dependencies" and
+  share it in Slack.
+
+This applies for really big ideas, and sometimes small ones, too! Check the [FAQ](#frequently-asked-questions) for
+more context. Writing a technical plan is the go-to solution for making technology decisions at Artsy because even
+if the decision is uncontroversial, the exercise of writing a plan will help guide your thought process _and_ will
+record the rationale for the decision. (This is historical context is super-userful.) You got this!
```
Comment (joeyAghion) [inline playbooks/technology-choices.md]:
```suggestion
record the rationale for the decision. (This historical context is super-userful.) You got this!
```

File: playbooks/technology-choices.md [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
 ---
 
 **High-level architecture and technology choices** are some of the most important and carefully considered
-decisions we make as engineers.
+decisions we make as engineers. This document goes into a lot of detail around how we made decisions, but the short
+version is:
+
+- If you have an idea,
+  [write a technical plan](https://www.notion.so/artsy/Technical-Plans-f94b206fcec54cee8b4d864e67d5b70f), then...
+- Share the plan with the team: bring it up
```
Comment (joeyAghion) [inline playbooks/technology-choices.md]:
```suggestion
- Share the plan with the wider engineering team: bring it up
```
(just to remove ambiguity of product vs. engineering team)

**Comment by ashfurrow** [comment]
@zephraph I've added a clarifying section here, and am curious what you think: https://github.com/artsy/README/pull/277/commits/a898bf596cd9f582648a2d9a8da09f182267736a

File: playbooks/technology-choices.md [hunk 3]
```diff
@@ -80,3 +92,26 @@ deliberately).
...
+
+#### When should I use an RFC? When should I write a technical plan?
+
+RFC, or _Requests for Comments_ are
+[defined in the RFC playbook](https://github.com/artsy/README/blob/master/playbooks/rfcs.md) as:
+
+> a process used in large open source orgs to coordinate talking about a change and giving many people the chance
+> to express ideas and discuss changes
+
+Technical were [introduced in #245](https://github.com/artsy/README/issues/245) as:
```
Comment (dleve123) [inline playbooks/technology-choices.md]:
```suggestion
Technical plans were [introduced in #245](https://github.com/artsy/README/issues/245) as:
```

File: playbooks/technology-choices.md [hunk 4]
```diff
@@ -80,3 +92,26 @@ deliberately).
...
+
+> an effort to make our planning more consistent, deliberate, and discoverable.
+
+These two things are not mutually exclusive, and there is some overlap. So which is the right one to use? In
+practice, either can work. These are both tools to help guide us, and shouldn't be seen as constraints.
+
+If you're asking which makes sense to use, this is a good general rule:
+
+- RFC's are better at discussing changes to _how_ we work.
+- Technical plans are better at discussing changes to _what_ tools we work with.
```
Comment (dleve123) [inline playbooks/technology-choices.md]:
```suggestion
- Technical plans are better at discussing changes to _what_ tools we work with and _how_ we plan on using tools and technologies to solution around a specific problem.

For technical plans that affect your team or a related team, you are _strongly encouraged_ to review and provide input on them.
```

File: playbooks/technology-choices.md [hunk 5]
```diff
@@ -80,3 +92,26 @@ deliberately).
...
+Technical were [introduced in #245](https://github.com/artsy/README/issues/245) as:
+
+> an effort to make our planning more consistent, deliberate, and discoverable.
+
+These two things are not mutually exclusive, and there is some overlap. So which is the right one to use? In
+practice, either can work. These are both tools to help guide us, and shouldn't be seen as constraints.
+
+If you're asking which makes sense to use, this is a good general rule:
+
+- RFC's are better at discussing changes to _how_ we work.
```
Comment (dleve123) [inline playbooks/technology-choices.md]:
```suggestion
- RFC's are better at discussing changes to _how_ we work. There is no expectation that any specific number of engineers needs to review an RFC and are such are very "opt in".
```

File: playbooks/technology-choices.md [hunk 6]
```diff
@@ -80,3 +92,26 @@ deliberately).
...
+- Technical plans are better at discussing changes to _what_ tools we work with.
+
+But there are always exceptions ([this RFC](https://github.com/artsy/README/issues/268), for example, covers both).
+You really can't go wrong, though!
```
Comment (dleve123) [inline playbooks/technology-choices.md]:
```suggestion
You really can't go wrong, though! A key point is to _write down your thinking_, ensuring that the team is communicating strongly about the topic at hand and allowing us to retrospect on decisions made.
```

**Pull request merged**